src/pybind/mgr/prometheus/module.py

```diff
@@ -593,6 +593,13 @@ class Module(MgrModule, OrchestratorClientMixin):
             desc='Do not include perf-counters in the metrics output',
             long_desc='Gathering perf-counters from a single Prometheus exporter can degrade ceph-mgr performance, especially in large clusters. Instead, Ceph-exporter daemons are now used by default for perf-counter gathering. This should only be disabled when no ceph-exporters are deployed.',
             runtime=True
+        ),
+        Option(
+            name='metrics_tls_enabled',
+            type='bool',
+            default=False,
+            desc='Enable TLS for the metrics endpoint',
+            runtime=True
```

Comment:
I'd recommend not adding this option here, plz have a look to the below comment.

Reply:
Removed. Added secure= monitoring stack variable.